examples/solids/elasticity.c

```diff
@@ -473,7 +473,7 @@ int main(int argc, char **argv) {
 
       // -------- Coarse KSP
       ierr = PCMGGetCoarseSolve(pc, &kspCoarse); CHKERRQ(ierr);
-      ierr = KSPSetType(kspCoarse, KSPPREONLY); CHKERRQ(ierr);
+      ierr = KSPSetType(kspCoarse, KSPCG); CHKERRQ(ierr);
```

Comment:
Curious to see perf, but we probably don't want this.

Reply:
It significantly drops the number of v-cycles in a SNES iteration in my initial tests. 16-14 to 9-8

Reply:
Yeah, at the expense of lots more global synchronization (so it'll scale much worse). The goal here will be to balance work happening on the coarse grid with that occurring in levels. It could be that richardson(2) on the GAMG part is a good balance. We can also make GAMG smoothers stronger. Alternatively, we could make our p-MG smoothers weaker so they don't over-solve.

Reply:
True. I dropped it for now so we can merge this. I can come back to this sort of tuning on the next PR.

Reply:
Cool. These knobs are also somewhat problem specific. We can write a tuning guide at some point, to teach users (and remind us) how to balance work through the MG cycle so don't waste flops on the fine grid and don't waste synchronization on the coarse grid.